app/controllers/concerns/administrate/punditize.rb

```diff
@@ -7,32 +7,58 @@ module Punditize
       included do
         private
 
+        def policy_namespaces
```

Comment:
Your reservation makes perfect sense! I've actually thought about this. Didn't want to use the singular `policy_namespace` because it has to return an array. I thought users might choose to return `Foo::Bar` if it is singular. I've refactored the code now to use the class method `pundit_policy_namespace` and users should set `Foo::Bar` instead of the array format `[:foo, :bar]`. I think the array is just an implementation detail from [Pundit](https://github.com/varvet/pundit/blob/5b22078621cf0249d39dc8c8bb0e037cae779fb0/lib/pundit/policy_finder.rb#L74) that we don't need to expose.

Reply:
Actually, I prefer having an overridable method like you had originally (but with the singular name). This is more in the spirit of "no DSLs" that we are trying to achieve with Administrate.

As for whether users should give lists of symbols `[:foo, :bar]` or straight modules `Foo::Bar`, I think it should be the list of symbols. When a user uses `Punditize`, they are already commited to Pundit, so there's no need to hide what's really going on. In fact it'll be easier for the user to understand (I think) that the namespace is `[:foo, :bar]` as it is what they would normally use with Pundit.

Does that make sense?

Reply:
> When a user uses `Punditize`, they are already commited to Pundit, so there's no need to hide what's really going on. In fact it'll be easier for the user to understand (I think) that the namespace is `[:foo, :bar]` as it is what they would normally use with Pundit.

Makes sense.